### PR TITLE
Refactor product detail layout

### PR DESCRIPTION
**Pull request by tvikito**
| Q             | A
| ------------- | ---
|Description, reason for the PR| we have very bad design on product detail page, this refactoring improves layout on this page, also we have small border-radius all over the app so the look is more modern
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://tv-fw-1640-refactor-product-detail-layout.odin.shopsys.cloud
  - https://cz.tv-fw-1640-refactor-product-detail-layout.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/components/Forms/Button/Button.tsx [hunk 1]
```diff
@@ -40,14 +38,13 @@ export const Button: FC<Props> = forwardRef(
         return (
             <button
                 className={twMergeCustom(
-                    'inline-flex w-auto cursor-pointer items-center justify-center gap-2 text-center font-bold uppercase outline-none transition-all hover:no-underline',
-                    size === 'small' ? 'py-1 px-4 text-sm' : 'py-3 px-8 text-base',
```
Comment (sebaholesz) [inline project-base/storefront/components/Forms/Button/Button.tsx:44]: Ask Martin Valenta about this. The original button comes from a styleguide we followed. I know it's a pretty small change and we shouldn't care, but just to be sure. You can close this comment as you wish...

File: project-base/storefront/components/Basic/Table/Table.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ type TableProps = {
 };
 
 export const Table: FC<TableProps> = ({ head, children, className }) => (
-    <div className={twMergeCustom('overflow-x-auto rounded-xl border-2 border-border p-6', className)}>
+    <div className={twMergeCustom('overflow-x-auto rounded border-2 border-border p-6', className)}>
```
Comment (sebaholesz) [inline project-base/storefront/components/Basic/Table/Table.tsx:9]: I have no problem with this change, as design is not something I focus on. IMHO Shopsys Platform could have zero design on SF, but still, there are stakeholders who care about how it looks. I just want to know if they're on board. Again, you can close this comment as you wish, just make sure this was discussed with Peta, Martin, maybe Vasek, Tom...

**Review by sebaholesz — CHANGES_REQUESTED** [review]

**Comment by sebaholesz** [comment]
This probably isn't where it belongs, but we have the scrollbar even here...
image

File: project-base/storefront/components/Forms/Button/Button.tsx (hunk 1 above)
Comment (tvikito) [reply project-base/storefront/components/Forms/Button/Button.tsx:44]: Yup, don't worry. Everyone will be notified about this new design.

File: project-base/storefront/components/Basic/Table/Table.tsx (hunk 2 above)
Comment (tvikito) [reply project-base/storefront/components/Basic/Table/Table.tsx:9]: Yup, don't worry. Everyone will be notified about this new design.

**Pull request merged**